src/index.js

```diff
@@ -1,3 +1,216 @@
-console.log(data);
+//Build doggy card
+const buildDogCard = (data) => {
+  const section = buildSection();
+  const sectionWithName = addH2(section, data.name);
+  const sectionWithImg = addImg(sectionWithName, data.image);
+  const sectionWithDiv = addDiv(sectionWithImg);
+  const sectionWithBio = addBio(sectionWithDiv, data.bio);
+  const sectionWithNaughty = addNaughty(sectionWithBio, data.isGoodDog);
+  const finalCard = addButton(sectionWithNaughty, data);
+  return finalCard;
+};
 
-// WRITE YOUR CODE BELOW!
+const buildSection = () => {
+  const newSection = document.createElement("section");
+  newSection.setAttribute("class", "main__dog-section");
+  return newSection;
+};
+
+const addH2 = (section, input) => {
+  const newH2 = document.createElement("h2");
+  newH2.innerText = input;
+  section.append(newH2);
+  return section;
+};
+
+const addImg = (section, img) => {
+  const newImg = document.createElement("img");
+  newImg.setAttribute("src", img);
+  section.append(newImg);
+  return section;
+};
+
+const addDiv = (section) => {
+  const newDiv = document.createElement("div");
+  newDiv.setAttribute("class", "main__dog-section__desc");
+  section.append(newDiv);
+  return section;
+};
+
+const addBio = (section, bio) => {
+  const newH3 = document.createElement("h3");
+  const newP = document.createElement("p");
+  const div = section.querySelector("div");
+  newH3.innerText = "Bio";
+  newP.innerText = bio;
+  div.append(newH3);
+  div.append(newP);
+  return section;
+};
+
+const addNaughty = (section, isGoodDog) => {
+  const newP = document.createElement("p");
+  const newEm = document.createElement("em");
+  isGoodDog ? (newP.innerText = " Yes") : (newP.innerText = " No");
+  newEm.innerText = "Is naughty?";
+  newP.setAttribute("id", "good-dog");
+  newP.prepend(newEm);
+  section.append(newP);
+  return section;
+};
+
+const addButton = (section, data) => {
+  const newButton = document.createElement("button");
+  data.isGoodDog ? newButton.innerText = "Bad Dog!" : newButton.innerText = "Good Dog!";
+  newButton.setAttribute("id", "good-button");
+  section.append(newButton);
+  return section;
+};
+
+//Builds header Buttons
+const buildHeaderButtons = (data) => {
+  const newLi = document.createElement("li");
+  newLi.setAttribute("class", "dogs-list__button");
+  newLi.innerText = data.name;
+  addEventListener(newLi, data);
+  return newLi;
+};
+
+const populateHeader = () => {
+  const header = document.querySelector(".dogs-list");
+  for (let i = 0; i < data.length; i++) {
+    let newButton = buildHeaderButtons(data[i]);
```

Comment:
This effectively duplicates all dogs in the header when adding a new one; only the new dog should be added

Reply:
Fixed!